internal/pkg/agent/storage/store/state_store.go

```diff
@@ -169,6 +170,18 @@ func (s *StateStore) SetAction(a fleetapi.Action) {
 	s.mx.Lock()
 	defer s.mx.Unlock()
 
+	if a == nil || reflect.ValueOf(a).IsNil() {
+		s.log.Debugf("trying to set an nil '%T' action, ignoring the action", a)
```

Comment:
Shouldn't this rather be an error log ? We're never gonna see this in logs collected via diagnostics otherwise...

Reply:
no, we're supposed to ignore `nil`. I'm just fixing the panic, not changing the behaviour . We have another issue to deal with the self-unenroll https://github.com/elastic/elastic-agent/issues/5428. We can address that there